DockService.Core/Services/IDockService.cs

```diff
@@ -0,0 +1,23 @@
+﻿using DockService.Core.Models;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DockService.Core.Services
+{
+    public interface IDockService
+    {
+        #region --required for persistance---
```

Comment:
It is better to define methods like DockShip(Ship ship).
This way you hide the implementation from the outside. Therefore other services that use an instance of `IDockService` do not need to know what exactly is happening.

Then you would not need to send event methods. This would be done in the `DockShip` implementation.